### DISCUSSION TITLE
Env vars assume direnv

### DISCUSSION THREAD

**Discussion by zimbatm** [numtide/prj-spec#2] [General]
Using env vars to point to the project root requires either using direnv or a wrapper script.

There is also a non-negligible risk that env vars might be lingering from another session and pointing to a different project root.

**Comment by zimbatm** [comment]
The alternative would be to change the spec so it looks for a common file like `.prj.toml` up the file hierarchy. That file might then contain all the configuration and metadata.

**Reply by Profpatsch** [reply]
wouldn’t depend on a file format, what about `.project/project-id.txt` and then have a plaintext string. The project root is the closest directory which contains a `.project` directory.
